packages/dds/tree/src/feature-libraries/sequence-field/rebase.ts

```diff
@@ -146,6 +165,78 @@ function rebaseMark<TNodeChange>(
             return clone(currMark);
         }
         default:
-            fail("Not implemented");
+            fail(`Unsupported mark type: ${baseType}`);
+    }
+}
+
+function handleCurrAttach<T>(
+    currMark: Attach<T>,
+    factory: MarkListFactory<T>,
+    lineageRequests: LineageRequest<T>[],
+    offset: number,
+) {
+    const rebasedMark = clone(currMark);
+    factory.pushContent(rebasedMark);
+    lineageRequests.push({ mark: rebasedMark, offset });
+}
+
+function isAttachAfterBaseAttach<T>(currMark: Attach<T>, baseMark: Attach<T>): boolean {
+    const lineageCmp = compareLineages(currMark.lineage, baseMark.lineage);
+    if (lineageCmp < 0) {
+        return false;
+    } else if (lineageCmp > 0) {
+        return true;
+    }
+
+    // TODO: Handle tiebreak
+    return false;
+}
+
+function compareLineages(
+    lineage1: LineageEvent[] | undefined,
+    lineage2: LineageEvent[] | undefined,
+): number {
+    if (lineage1 === undefined || lineage2 === undefined) {
+        return 0;
+    }
+
+    const lineage1Offsets = new Map<RevisionTag, number>();
+    for (const event of lineage1) {
+        lineage1Offsets.set(event.revision, event.offset);
+    }
+
+    for (let i = lineage2.length - 1; i >= 0; i--) {
```

Comment:
It would be nice to add a comment explaining that, while two marks could bear unrelated lineage information for a given revision (let's call it "rev Foo") (meaning each mark was affected by independent detaches at that revision), the fact that we're walking through the lineage from most recent to most ancient lineage nodes ensures we won't ever compare the indices of such nodes because in order for two marks to end up in the same gap despite having such unrelated lineage nodes from the same revision (rev Foo), they would have needed to be affected by a deletion that merged the independent gaps that they were targeting, and we'll encounter the lineage nodes for that revision before we encounter the lineage nodes for rev Foo.
Hopefully you can find a cleaner way to say all that, but I think it's not obvious from the code that it works at it should because of these facts.

Also, it would be nice to add a test with such a rev Foo.